core/modules/server/server.js

```diff
@@ -196,25 +196,33 @@ Server.prototype.requestHandler = function(request,response) {
 		response.end();
 		return;
 	}
-	// Set the encoding for the incoming request
-	// TODO: Presumably this would need tweaking if we supported PUTting binary tiddlers
-	request.setEncoding("utf8");
-	// Dispatch the appropriate method
-	switch(request.method) {
-		case "GET": // Intentional fall-through
-		case "DELETE":
+
+	//receive the request body if necessary and hand off to the route handler
+	if(route.bodyFormat === "stream" || request.method === "GET" || request.method === "HEAD"){
```

Comment:
Shouldn't we check for "DELETE" too here?

Reply:
Good question. I looked it up, and according to [this answer on Stack Overflow](https://stackoverflow.com/a/51112818/258482) the HTTP spec allows all methods except TRACE to have a request body.